### DISCUSSION TITLE
v1.7.8

### DISCUSSION THREAD

**Discussion by aluminum-ice** [aluminum-ice/pwnagotchi#41] [Announcements]
- 84fb9a0cac0a9a9cba22e26b28798847bff3846a Re-enabled installing the latest version of nexmon
- 59a61e32cdabda692b3bc009c79c235d1c5cee0f Added lnav to the apt-get packages to install
- ab36f2c5fc810657fb3a4c6e9d9c261da8a38a5e Added logging.exception to catch generic errors when opening a web socket
- 383d71c1fd5986a61412cedb9fdc0c08e3df4d89 Increased max retries for requests to 5, added a loop to repeat a failed request twice with a 1-s timeout, and added try+exception to catch errors
- 3535529ee8115b9bb7fe0e4e9a8d363d02d6f7a7 Fixed ephemeral port exhaustion
- 5e017d0650c9c5d0f99253f075ab173d11719e6a Removed unneeded folders to save space

**Full Changelog**: https://github.com/aluminum-ice/pwnagotchi/compare/v1.7.7...v1.7.8

<hr /><em>This discussion was created from the release <a href='https://github.com/aluminum-ice/pwnagotchi/releases/tag/v1.7.8'>v1.7.8</a>.</em>

**Comment by wpa-2** [comment]
I have a strange issue on this build when placing custom plugins in /usr/local/share/pwnagotchi/custom-plugins they are not seen by the UI or work but IIf i move them to /usr/local/lib/python3.7/dist-packages/pwnagotchi/plugins/default they work right away! 
100% Didnt have this on 1.7.7

**Reply by aluminum-ice** [reply]
Do you have the right location set in your config file? Note that I install the plugin's I make part of the image into /usr/local/share/pwnagotchi/custom-plugins/. I didn't touch any of that code so it should not have broken.

**Comment by ryanbeard82** [comment]
Pardon the noob question, but is there any way to update directly from the previous 1.7.7 build or does this require a fresh install on the sd card?

**Reply by ryanbeard82** [reply]
nvm - replacing auto-update.py with the latest then forcing an update check (`sudo rm -rf /root/.auto-update && sudo service pwnagotchi restart`) worked perfectly.

**Reply by ch4r13** [reply]
I just enabled plugin auto_update and it found new version and downloaded and installed it. Everything automatically :) Perfect !
EDIT: I updated from 1.7.7 ver.

**Comment by wpa-2** [comment]
> auto-update.py
Did you mean auto-update.py file from 1.7.8 ?

**Reply by ryanbeard82** [reply]
No. I was on 1.7.7 previously and was looking for a way to update to 1.7.8 without requiring a fresh install. Copying the auto-update.py contents from 1.7.8 and pasting into my 1.7.7 version (or just update the list of repos monitored for update) resolved the issue, resulting in auto update.

**Reply by wpa-2** [reply]
That's what I asked did you use the auto update file from 1.7.8

**Reply by wpa-2** [reply]
Thabsk for clarification though will try that.

**Comment by aluminum-ice** [comment]
> No. I was on 1.7.7 previously and was looking for a way to update to 1.7.8 without requiring a fresh install. Copying the auto-update.py contents from 1.7.8 and pasting into my 1.7.7 version (or just update the list of repos monitored for update) resolved the issue, resulting in auto update.

Glad its working. In the future, that should enable you to auto-update.

**Comment by ryanbeard82** [comment]
> Thabsk for clarification though will try that.
Sorry, I misunderstood the question.

You're correct. Just copy the contents from the auto-update.py in the 1.7.8 release, and paste into your existing auto-update.py file. After that, delete the update bookmark file and restart the pwnagotchi service (`sudo rm -rf /root/.auto-update && sudo service pwnagotchi restart`). My device was updated within minutes after restarting.

**Comment by APAAlpha** [comment]
Hey guys, the file is a 7z, how do i flash this the my SD card? all the imagers ik dont support that?

**Reply by wpa-2** [reply]
Raspberry pi imager does or extract it first. 
Google is your friend.
